### ISSUE TITLE
llvm/IR/DebugLoc.h: No such file or directory

### ISSUE THREAD

**Issue by stugol** [manastech/crystal#2179]
I'm probably doing something stupid again, but when I try to compile Crystal now, I get:
```
g++ -c -o src/llvm/ext/llvm_ext.o src/llvm/ext/llvm_ext.cc `/usr/bin/llvm-config --cxxflags`
src/llvm/ext/llvm_ext.cc:2:30: fatal error: llvm/IR/DebugLoc.h: No such file or directory
compilation terminated.
make: *** [src/llvm/ext/llvm_ext.o] Error 1
```
The entire sequence of build commands I'm using is as follows, from a fresh Debian 64-bit installation:
```
apt-get update
apt-get install -y llvm libpcre3-dev build-essential curl git zlib1g-dev automake libtool pkg-config unp libedit-dev libssl-dev libevent-dev libgc-dev libreadline-dev libgdbm-dev
curl http://dist.crystal-lang.org/apt/setup.sh | bash
apt-get install -y crystal
git clone https://github.com/manastech/crystal.git
cd crystal

git clone git://github.com/ivmai/bdwgc.git
cd bdwgc
git clone git://github.com/ivmai/libatomic_ops.git
autoreconf -vif
automake --add-missing
./configure
make
make install

curl -O http://www.xmailserver.org/pcl-1.12.tar.gz
unp pcl-1.12.tar.gz
cd pcl-1.12
./configure
make
make install
cd ../..

rm /usr/lib/libgc.so* || true
apt-get remove libpcl1-dev || true

make
cd ..
ln -s /root/crystal/bin/crystal /usr/local/bin/
ln -s /usr/lib/x86_64-linux-gnu/libcrypto.a /opt/crystal/embedded/lib/
ln -s /usr/lib/x86_64-linux-gnu/libz.a /opt/crystal/embedded/lib/

echo >>/etc/bash.bashrc export LIBRARY_PATH=/usr/local/lib:$LIBRARY_PATH
echo >>/etc/bash.bashrc export LD_LIBRARY_PATH=/usr/local/lib
. /etc/bash.bashrc 
```
I've tried also installing `llvm-dev`. It didn't help.

**Comment by kirbyfan64** [comment]
What's your version of LLVM?

**Comment by stugol** [comment]
```
# apt-get install llvm
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following extra packages will be installed:
  binfmt-support libffi-dev libllvm3.0 llvm-3.0 llvm-3.0-dev llvm-3.0-runtime llvm-runtime
Suggested packages:
  llvm-3.0-doc
The following NEW packages will be installed:
  binfmt-support libffi-dev libllvm3.0 llvm llvm-3.0 llvm-3.0-dev llvm-3.0-runtime llvm-runtime
0 upgraded, 8 newly installed, 0 to remove and 91 not upgraded.
```

**Comment by asterite** [comment]
You need LLVM 3.6

**Comment by stugol** [comment]
Well, that's very odd. Because I currently have a system running 3.5.0 that can compile Crystal (and Crystal code) perfectly:
```
# llvm-config --version
3.5.0
# crystal build linux-setup.cr
Using compiled compiler at .build/crystal
```
So....I guess it _doesn't_ need 3.6?

**Comment by kirbyfan64** [comment]
I think both 3.5 and 3.6 work, but 3.0 is definitely *way* too old, and 3.7 doesn't work.

**Comment by stugol** [comment]
Debian 7 doesn't have an llvm package after 3.0, but I'm sure I've compiled Crystal recently in Debian 7. Like, a week or two ago. When did Crystal start requiring >3.0?

**Comment by kirbyfan64** [comment]
...over a year ago...

**Comment by stugol** [comment]
Well, that is definitely strange. Anyhow, I've figured out how to get llvm 3.5 on Debian 7, so hopefully it'll work this time. Please leave this issue open for a little while in case it doesn't work.

**Comment by stugol** [comment]
How can I programmatically detect if the currently-installed version of Crystal is the newest one? Can I query `git` in some way, and compare it with `crystal -v`?

**Comment by kirbyfan64** [comment]
@stugol Check if `crystal -v | sed 's/.*\[\(.*\)\].*/\1/'` is equal to `git rev-parse HEAD | sed 's/\(.......\).*/\1/'`.

**Comment by stugol** [comment]
I'd have to clone the repo first though, right? Is there no way to tell otherwise?

**Comment by stugol** [comment]
How much memory do I need to compile Crystal? Because here's an excerpt from my auto-build logs:
```
00:00:25     g++ -c -o src/llvm/ext/llvm_ext.o src/llvm/ext/llvm_ext.cc `/usr/bin/llvm-config-3.5 --cxxflags`
00:00:27     cc    -c -o src/ext/sigfault.o src/ext/sigfault.c
00:00:27     ar -rcs src/ext/libcrystal.a src/ext/sigfault.o
00:00:27     CRYSTAL_CONFIG_PATH=`pwd`/src ./bin/crystal build  -o .build/crystal src/compiler/crystal.cr
00:09:50     fork: Cannot allocate memory
00:09:50     [5626458] ???
00:09:50     [7894280] ???
00:09:50     [9427530] ???
00:09:50     [9417849] ???
00:09:50     [7942180] ???
00:09:50     [12326686] ???
00:09:50     [12324459] main +23387
00:09:50     [47067660689069] __libc_start_main +253
00:09:50     [5620497] ???
00:09:50
00:09:50     Error: Error: you've found a bug in the Crystal compiler. Please open an issue, including source code that will allow us to reproduce the bug: https://github.com/manastech/crystal/issues
00:09:51     Error: make: *** [.build/crystal] Error 1
00:09:51   Error: `make` returned 2
```

**Comment by ysbaddaden** [comment]
Closing since initial issue is resolved (only LLVM 3.5 or 3.6 are supported).
